docs/install.md

```diff
@@ -0,0 +1,110 @@
+
+* If you are using a self managed cluster, ensure the flag `--allow-privileged=true` for `kube-apiserver`.
+
+* Important: If you intend to use the Volume Snapshot feature, the [Kubernetes Volume Snapshot CRDs](https://github.com/kubernetes-csi/external-snapshotter/tree/master/client/config/crd) must be installed **before** the FSx for OpenZFS CSI driver. For installation instructions, see [CSI Snapshotter Usage](https://github.com/kubernetes-csi/external-snapshotter#usage).
+
+## Installation
+### Set up driver permissions
+The driver requires IAM permission to talk to the Amazon FSx for OpenZFS service to create/delete file systems, volumes, and snapshots on user's behalf. There are several methods to grant driver IAM permission:
+
+There are several methods to grant the driver IAM permissions:
+* Using [IAM roles for ServiceAccounts](https://docs.aws.amazon.com/eks/latest/userguide/iam-roles-for-service-accounts.html) (**Recommended**) - Create a Kubernetes service account for the driver and attach the AmazonFSxFullAccess AWS-managed policy to the service account with the following command. If your cluster is in the AWS GovCloud (US-East) or AWS GovCloud (US-West) AWS Regions, then replace arn:aws: with arn:aws-us-gov:
+```sh
+eksctl create iamserviceaccount \
+    --name fsx-openzfs-csi-controller-sa \
+    --namespace kube-system \
+    --cluster $cluster_name \
+    --attach-policy-arn arn:aws:iam::aws:policy/AmazonFSxFullAccess \
+    --approve \
+    --role-name AmazonEKSFSxOpenZFSCSIDriverFullAccess \
+    --region $region_code
+```
+
+* Using IAM [instance profile](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_use_switch-role-ec2_instance-profiles.html) - create the following IAM policy and attach the policy to the instance profile IAM role of your cluster's worker nodes. 
+See [here](https://docs.aws.amazon.com/eks/latest/userguide/create-node-role.html) for guidelines on how to access your EKS node IAM role.
+```sh
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Effect": "Allow",
+      "Action": [
+        "iam:CreateServiceLinkedRole",
+        "iam:AttachRolePolicy",
+        "iam:PutRolePolicy"
+       ],
+      "Resource": "arn:aws:iam::*:role/aws-service-role/fsx.amazonaws.com/*"
+    },
+    {
+      "Action":"iam:CreateServiceLinkedRole",
+      "Effect":"Allow",
+      "Resource":"*",
+      "Condition":{
+        "StringLike":{
+          "iam:AWSServiceName":[
+            "fsx.amazonaws.com"
+          ]
+        }
+      }
+    },
+    {
+      "Effect": "Allow",
+      "Action": [
+        "fsx:CreateFileSystem",
+        "fsx:DeleteFileSystem",
+        "fsx:DescribeFileSystems",
+        "fsx:CreateVolume",
+        "fsx:DeleteVolume",
+        "fsx:DescribeVolumes",
+        "fsx:CreateSnapshot",
+        "fsx:DeleteSnapshot",
+        "fsx:DescribeSnapshots",
+        "fsx:TagResource"
+      ],
+      "Resource": ["*"]
+    }
+  ]
+}
+```
+
+### Configure driver toleration settings
```

Comment:
From my understanding this is a very common use case. Many users with complex workloads/architecture don't want the CSI driver deployed to all of the nodes in their cluster due to resource/memory utilization. Especially since we don't have a min base image prepared yet.